File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        vector<int>ans(2);
        map<int,int>map;
        for(int i=0;i<nums.size();i++){
            if(map.contains(target-nums[i])){
                ans[0]=map.at(target-nums[i]);
                ans[1]=i;
                return ans;
            }
            map.insert({nums[i],i});
        }
        return ans;
    }
};

File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        my_map = {}
        for i in range(len(nums)):
            req = target - nums[i]
            if req in my_map:
                return [my_map[req], i]
            my_map[nums[i]] = i

        return []
